.buildbot.sh

```diff
@@ -0,0 +1,110 @@
+WEBKIT="$(realpath "$PWD")"
+
+BUILDBOT_SCRATCH=$(mktemp -d --tmpdir -t .buildbot-scratch-XXXXXX)
+echo "Using temporary directory: $BUILDBOT_SCRATCH"
+cleanup() {
+  echo "Cleaning up: $BUILDBOT_SCRATCH"
+  rm -rf "$BUILDBOT_SCRATCH"
+  trap '' INT TERM EXIT
+  exit
+}
+trap cleanup INT TERM EXIT
+
+# We need to run ./cheribuild.py, but it needs a source tree. $CHERIBASE is
+# provided (with at least the SDK and filesystem images), but is read-only, so
+# we need to copy it.
+#
+# TODO: Consider using symlinks to sub-trees that only need to be read-only.
+CHERIRW="$BUILDBOT_SCRATCH/cheri"
+echo "Copying $CHERIBASE to $CHERIRW..."
+RULES=(
+  '--include=/output/'
+  '--include=/icu/'     # ICU probably isn't present, but we'll take it if it is.
+  '--exclude=/*/'
+  '--exclude=*.img'     # Don't copy large disk images.
+  # Files in /rescue are hardlinked together and end up very large after
+  # copying. -H could preserve links, but it's easier to just exclude them.
+  '--exclude=**/rootfs-*/rescue/'
+)
+rsync -a --chmod=u+w "$CHERIBASE/" "$CHERIRW" "${RULES[@]}"
+
+echo "Symlinking WebKit test subject into the CHERI root: $CHERIRW/webkit -> $WEBKIT"
+ln -fsT "$WEBKIT" "$CHERIRW/webkit"
+
+TARGET_FILES="$BUILDBOT_SCRATCH/target_files"
+mkdir -p "$TARGET_FILES"/builds
+cp -r -t "$TARGET_FILES" "PerformanceTests/SunSpider/tests/sunspider-1.0.1"
+cp .buildbot-test.sh "$TARGET_FILES"/test.sh
+chmod +x "$TARGET_FILES"/test.sh
+
+build() {
+  ARCH="$1"
+  shift
+  # Flatten the build configuration to form a unique destination directory.
+  local FLAT=$(echo "$*" |
+               sed "s/\(.\+\)/\L\1/g;       # Lower case
+                    s/--morello-webkit\///g;
+                    s/\s\+/_/g;
+                    s/[^a-z0-9_-]//g;"
+  )
+  local OUTDIR="$(realpath "$TARGET_FILES"/builds/$FLAT)"
+  pushd "$CHERIBUILD"
+
+  # We use 'script' (below) to make cheribuild think that it has an interactive
+  # shell. This isn't ideal, but cheribuild otherwise hangs with SIGTTOU on
+  # consecutive calls: https://github.com/CTSRD-CHERI/cheribuild/issues/182
+  # TODO: Once the cheribuild bug is fixed, remove this workaround.
+
+  cheribuild() {
+    echo "Building: ./cheribuild.py --source-root=\"$CHERIRW\" --skip-update $*"
+    script -qec "./cheribuild.py --source-root=\"$CHERIRW\" --skip-update $*" /dev/null
+  }
+
+  cheribuild "icu4c-native"
+  cheribuild "icu4c-$ARCH"
+  # --clean and/or --reconfigure are sometimes required when building different
+  # configurations in sequence.
+  cheribuild --clean --reconfigure morello-webkit-$ARCH "$@"
+
+  local FSROOT="$CHERIRW"/output/rootfs-$ARCH
+
+  echo "Build complete."
+  mkdir "$OUTDIR"
+  mkdir "$OUTDIR/bin"
+  mkdir "$OUTDIR/lib"
+  cp -a -t "$OUTDIR/bin" "$FSROOT"/opt/$ARCH/webkit/bin/*
+  cp -a -t "$OUTDIR/lib" "$FSROOT"/opt/$ARCH/webkit/lib/*
+  cp -a -t "$OUTDIR/lib" "$FSROOT"/usr/local/$ARCH/lib/libicu*
+  popd
+}
+
+export PYTHONPATH="$CHERIBUILD"/test-scripts
```

Comment:
I need to remove this (as advised by @0152la).

Reply:
Actually it's useful to keep this here because it means it can run standalone. Will that be a problem? I could set it only if it's not already set, but that might be confusing if it's set to something unhelpful.

Reply:
Setting it when it's not set is reasonable. The idea behind removing it was that the repos should not expect the buildbot instance to have things in specific places.